subprojects/logging/src/main/java/org/gradle/internal/logging/progress/DefaultProgressLoggerFactory.java

```diff
@@ -40,25 +41,38 @@ public ProgressLogger newOperation(Class loggerCategory) {
         return newOperation(loggerCategory.getName());
     }
 
+    public ProgressLogger newOperation(Class loggerCategory, OperationIdentifier operationIdentifier) {
```

Comment:
should be typed `Object` and I would call the parameter something like "buildOperationIdentifier" to make it clear whether it refers to a build operation id or a progress operation id.